### ISSUE TITLE
Compile failed due to not found libcache.h

### ISSUE THREAD

**Issue by qianguozheng** [ntop/ntopng#1510]
I update nDPI to the latest(2476c3c1b63b87ae85254f717315b9f8a52b4f7e), and update ntopng to the latest(4e4020a480d91c1f921123ebf3824b87751ce636).
But still cannot make ntopng being compiled ok.

below is the error not found libcache.h.

how should I do to make it work fine ?
```
g++ -g -Wall -I/home/weeds/magicwifi/test/ntopng -I/home/weeds/magicwifi/test/ntopng/include -I/usr/local/include -D_FILE_OFFSET_BITS=64 -I/usr/include/hiredis -I/usr/include/hiredis -I/home/weeds/magicwifi/test/ntopng/third-party/mongoose -I/usr/local/include/json-c  -I/usr/local/include/libndpi-2.1.0/libndpi -I/home/weeds/magicwifi/test/ntopng/third-party/LuaJIT-2.1.0-beta3/src  -I/home/weeds/magicwifi/test/ntopng/third-party/zeromq-4.1.3/include -I/usr/include/mysql  -I/home/weeds/magicwifi/test/ntopng -I/home/weeds/magicwifi/test/ntopng/include -I/usr/local/include -I/home/weeds/magicwifi/test/ntopng/third-party/http-client-c/src/  -I/usr/include/openssl  -DDATA_DIR='"/usr/local/share"' -I/home/weeds/magicwifi/test/ntopng/third-party/libgeohash -I/home/weeds/magicwifi/test/ntopng/third-party/patricia   -c src/Geolocation.cpp -o src/Geolocation.o
In file included from /usr/local/include/libndpi-2.1.0/libndpi/ndpi_main.h:30:0,
                 from /home/weeds/magicwifi/test/ntopng/include/ntop_includes.h:107,
                 from src/Geolocation.cpp:29:
/usr/local/include/libndpi-2.1.0/libndpi/ndpi_typedefs.h:28:49: fatal error: ../lib/third_party/include/libcache.h: No such file or directory
compilation terminated.
Makefile:154: recipe for target 'src/Geolocation.o' failed
make: *** [src/Geolocation.o] Error 1
```

**Comment by emanuele-f** [comment]
Hi, you need to clone ndpi into `/home/weeds/magicwifi/test` and compile it. Please execute:
```
cd /home/weeds/magicwifi/test
git clone https://github.com/ntop/nDPI.git
cd nDPI
./autogen.sh && ./configure && make
cd ../ntopng
./autogen.sh && ./configure && make
``` 

You should be able to compile it now.

**Comment by qianguozheng** [comment]
I follow your instruction, but still fail.

and I think it should be the relative path of libcache.h problem, actually I don't have any path like this *../lib/third_party/include/libcache.h: No such file or directory*, this is the problem.


```
 2143  cd /home/weeds/magicwifi/test/
 2144  rm -rf nDPI/
 2145  cd /home/weeds/magicwifi/test/
 2146  git clone https://github.com/ntop/nDPI.git
 2147  cd nDPI/
 2148  ./autogen.sh && ./configure && make
 2149  cd ../ntopng/
 2150  ./autogen.sh && ./configure && make

g++ -g -Wall -I/home/weeds/magicwifi/test/ntopng -I/home/weeds/magicwifi/test/ntopng/include -I/usr/local/include -D_FILE_OFFSET_BITS=64 -I/usr/include/hiredis -I/usr/include/hiredis -I/home/weeds/magicwifi/test/ntopng/third-party/mongoose -I/usr/local/include/json-c  -I/usr/local/include/libndpi-2.1.0/libndpi -I/home/weeds/magicwifi/test/ntopng/third-party/LuaJIT-2.1.0-beta3/src  -I/home/weeds/magicwifi/test/ntopng/third-party/zeromq-4.1.3/include -I/usr/include/mysql  -I/home/weeds/magicwifi/test/ntopng -I/home/weeds/magicwifi/test/ntopng/include -I/usr/local/include -I/home/weeds/magicwifi/test/ntopng/third-party/http-client-c/src/  -I/usr/include/openssl  -DDATA_DIR='"/usr/local/share"' -I/home/weeds/magicwifi/test/ntopng/third-party/libgeohash -I/home/weeds/magicwifi/test/ntopng/third-party/patricia   -c src/Geolocation.cpp -o src/Geolocation.o
In file included from /usr/local/include/libndpi-2.1.0/libndpi/ndpi_main.h:30:0,
                 from /home/weeds/magicwifi/test/ntopng/include/ntop_includes.h:107,
                 from src/Geolocation.cpp:29:
/usr/local/include/libndpi-2.1.0/libndpi/ndpi_typedefs.h:28:49: fatal error: ../lib/third_party/include/libcache.h: No such file or directory
compilation terminated.
Makefile:154: recipe for target 'src/Geolocation.o' failed
make: *** [src/Geolocation.o] Error 1
```

**Comment by qianguozheng** [comment]
I make it work by create a folder of  `lib/third_party/include/` under /usr/local/libndpi-xxx, and I think this problem maybe caused by that I have already install nDPI before, but not uninstall before execute your instruction, and I modify the Makefile of ntopng to make it find the libndpi.a at where I compile nDPI.

```
#HAS_NDPI=$(shell pkg-config --exists libndpi; echo $$?)
#ifeq ($(HAS_NDPI), 0)
#    NDPI_INC = $(shell pkg-config --cflags libndpi | sed -e 's/\(-I[^ \t]*\)/\1\/libndpi/g')
#    NDPI_LIB = $(shell pkg-config --libs libndpi)
#    NDPI_LIB_DEP =
#else
    NDPI_INC=-I../nDPI/src/include
    NDPI_LIB=../nDPI/src/lib/.libs/libndpi.a
    NDPI_LIB_DEP=$(NDPI_LIB)
#endif
```

Thanks.
